### ISSUE TITLE
markdown language is invisible and unmodifiable in theme "deep black"

### ISSUE THREAD

**Issue by NNRepos** [notepad-plus-plus/notepad-plus-plus#9649]
### Description of the Issue

when editing .md files, and using the "deep black" theme, both the text and the background of the current line are black, and therefore the text is rendered invisible. I tried modifying the language under the style configurator, but alas, it is not present.

### Steps to Reproduce the Issue

1. use deep black theme
2. open .md file with text
3. witness true terror

### Expected Behavior

text that can be seen
Markdown can be modified

### Actual Behavior

text that cannot be seen
Markdown cannot be modified

thx

**Comment by thomassmith311** [comment]
The problem seems to be occurring with all the themes, though, not just Deep Black.

**Comment by mere-human** [comment]
I agree, this has to be fixed. I wanted to do this someday, but lately I don't have enough time.
Usually, I just set language to "Normal text" for *.md and use this plugin  https://github.com/mohzy83/NppMarkdownPanel

**Comment by dodmi** [comment]
This can't be fixed easily. 
- Markdown support comes in form of a "custom language file"
- While "standard languages" can be defined per theme (in the theme files), custom language files lack the support of per theme definitions
- So if you adapt the MD language file to fit dark themes, it propably wouldn't produce readable output in bright themes, including the default theme

You can define a second custom mark down language file with your own color definitions. Play around with the file name until it takes precedence over the included one (this is also ugly, but doesn't need manual action, once it is done).

**Comment by mere-human** [comment]
> * While "standard languages" can be defined per theme (in the theme files), custom language files lack the support of per theme definitions

I wonder what do we need to make the markdown files "standard"? Define a custom Scintilla lexer? I'm not very familiar with this area (yet).

**Comment by Ekopalypse** [comment]
Defining a custom language with the [colorStyle ](https://npp-user-manual.org/docs/user-defined-language-system/#udl-and-themes) tag solves all problems related to different foreground and background colors.
Changing the other Markdown colors must be done via the UDL dialog.

**Comment by vinsworldcom** [comment]
See the [community discussion](https://community.notepad-plus-plus.org/topic/19510/markdown-lexer).  Scintilla already provides as Markdown Lexer, Notepad++ just doesn't use it.

Cheers.

**Comment by vinsworldcom** [comment]
Note you can add:

`colorStyle="1" `

to your UDL files to make the background color "transparent".  In my Markdown UDL file:

```
...
        <Styles>
            <WordsStyle name="DEFAULT" fgColor="E0E2E4" bgColor="FFFFFF" colorStyle="1" fontName="" fontStyle="0" nesting="0" />
            <WordsStyle name="COMMENTS" fgColor="66747B" bgColor="FFFFFF" colorStyle="1" fontName="" fontStyle="0" nesting="0" />
            <WordsStyle name="LINE COMMENTS" fgColor="66747B" bgColor="FFFFFF" colorStyle="1" fontName="" fontStyle="0" nesting="0" />
            <WordsStyle name="NUMBERS" fgColor="FFCD22" bgColor="FFFFFF" colorStyle="1" fontName="" fontStyle="0" nesting="0" />
...
```

You'll notice my background color is defined as "FFFFFF" or white, yet I use a dark theme and this is how it renders:

![image](https://user-images.githubusercontent.com/9657169/117062189-2606ae80-acf1-11eb-96e0-05646baa62fe.png)


Note, you can also achieve this effect in the GUI when creating a UDL by right-clicking the color you want to create a cross-hatch pattern over the color:

![image](https://user-images.githubusercontent.com/9657169/117062517-81d13780-acf1-11eb-9dd2-1f66b1a6a0e1.png)

You'll notice without the cross hatch, the background color "shines" through:

![image](https://user-images.githubusercontent.com/9657169/117062619-9f9e9c80-acf1-11eb-8d09-e412a3520481.png)

Hope this helps.

Cheers.

**Comment by NNRepos** [comment]
> See the [community discussion](https://community.notepad-plus-plus.org/topic/19510/markdown-lexer). Scintilla already provides a Markdown Lexer, Notepad++ just doesn't use it.
> 
> Cheers.

are you planning on supporting the markdown lexer in npp? 
if so, can you link the issue/PR # here? maybe i could help.

**Comment by vinsworldcom** [comment]
@NNRepos no, the Notepad++ author does not use the Scintilla Markdown lexer and instead has chosen to do Markdown lexing through a UDL.  If you really need the Scintilla Markdown lexer working in Notepad++, there are ways to get PythonScript to activate it, but it will never be seamless in that you'll never be able to configure the coloring through Style Configurator unless it is properly integrated - I don't count on that ever happening.

Cheers.

**Comment by ArkadiuszMichalski** [comment]
Feature request: Markdown support https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8083.
LexMarkdown vs UDL Markdown https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11099

I don't know what is missing in Markdown Lexer compared with UDL, but if someone has time then can try to work around this lexer: https://github.com/ScintillaOrg/lexilla/blob/master/lexers/LexMarkdown.cxx

If the result will be comparable or better, it will probably be switched to the built-in one.
